plugins/modules/s3_object_info.py

```diff
@@ -0,0 +1,614 @@
+#!/usr/bin/python
+# This file is part of Ansible
+# GNU General Public License v3.0+ (see COPYING or https://wwww.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+
+DOCUMENTATION = r'''
+---
+module: s3_object_info
+version_added: 5.0.0
+short_description: Gather information about objects in S3
+description:
+    - Describes objects in S3.
+author:
+  - Mandar Vijay Kulkarni (@mandar242)
+options:
+  bucket_name:
+    description:
+      - The name of the bucket that contains the object.
+    required: true
+    type: str
+  object_name:
+    description:
+      - The name of the object.
+    required: false
+    type: str
+  object_attributes:
```

Comment:
I don't think we need to have two `object_attributes` def We can probably move this one inside object details to overwrite https://github.com/ansible-collections/amazon.aws/pull/977/files#diff-c19ed6d5998f0666b7a49cc21f1da8909f61a4c66f4a8835fd2704ab3542126aR49

Reply:
Renamed to `attributes_list` and moved inside `object_details`.

```
- name: Retrieve detailed S3 object information
  amazon.aws.s3_object_info:
    bucket_name: MyTestBucket
    object_name: MyTestObject
    object_details:
      object_acl: true
      object_attributes: true
      attributes_list:
        - ETag
        - ObjectSize
        - StorageClass

```